src/Ember/Client/index.ts

```diff
@@ -388,24 +394,24 @@ export class EmberClient extends EventEmitter<EmberClientEvents> {
 			const i = pathArr.shift()
 			if (!i) break // TODO - this will break the loop if the path was `1..0`
 			if (!tree) break
-			let next = getNextChild(tree, i)
-			if (!next) {
-				const req = await this.getDirectory(tree)
-				tree = (await req.response) as NumberedTreeNode<EmberElement>
-				next = getNextChild(tree, i)
-			}
-			tree = next
+			const req = await this.getDirectory(tree)
+			tree = (await req.response) as NumberedTreeNode<EmberElement>
+			tree = getNextChild(tree, i)
```

Comment:
Can you help me understand why this is needed in regards to the main issue this PR addresses?

The reason I am asking it makes it such that every time a library user uses this call the library will fetch all nodes in the path even though we have already fetched these before and can just look at them locally.